### ISSUE TITLE
Rotated files date

### ISSUE THREAD

**Issue by bragma** [pm2-hive/pm2-logrotate#17]
I'm not sure what's the rational behind the rotated file names. Actual logic is to take the current date and subtract 1 of the specified time unit. As an example, if today is 2015-12-01 and interval_unit is 'MM', the file will be named 2015-11-01.
This may make some sense if the logrotate module is running continuously and rotation is time based only, but files can also be rotated on size. Suppose I have a max_size of 1K, interval_unit is 'MM' and my apps produce 1K of logs per minute. The logs generated today will be rotated due to size, but the file name will be 2015-11-01 while the content is that of the last minute of today.
There is also a problem with the file deletion system if the unit is changed. Suppose you have an interval_unit of 'DD' and a retention of 2, then suppose 2 files are created today, with the names:

```
somelog-out__2015-11-30-15-00.log
somelog-out__2015-11-30-15-05.log
```

(notice the date of yesterday, which is today minus 1 day)

Now, suppose I change the interval_unit to 'MM'. If a new file is created, the file will have a name like:

```
somelog-out__2015-11-01-15-10.log
```

(notice the date of 1 month ago, which is today minus 1 month)

The log rotation system will sort the files in this way:

```
somelog-out__2015-11-30-15-00.log
somelog-out__2015-11-30-15-05.log
somelog-out__2015-11-01-15-10.log <<< NEWEST FILE&CONTENT, OLDEST NAME
```

And then will delete what thinks is the OLDEST file, i.e. "somelog-out__2015-11-01-15-10.log". But this file contains the latest log lines instead...

My suggestions are:

1. allow to name the file with the actual date (the date when the file is rotated).
2. sort files for deletion using file date, not their names

**Comment by bragma** [comment]
Added an option to specify the file naming criteria in PR https://github.com/pm2-hive/pm2-logrotate/pull/16

**Comment by vmarchaud** [comment]
Fixed in latest update (v2.0.0).
